### ISSUE TITLE
[BRUKERHISTORIE] Slette katalog ved sletting av høsteendepunkt

### ISSUE THREAD

**Issue by oystein-asnes** [Informasjonsforvaltning/behov#802]
Som: datatilbyder
Ønsker jeg at: katalogen slettes når jeg sletter et høstendepunkt
Slik at:  test-data fjernes når jeg er ferdig eller ønsker å begynne på nytt med tom katalog

Relatert til epos: 

I dag kan bruker slette høsteendepunktet i demomiljøet, men katalogen blir liggende igjen. Brukeren mister da skrive-rettigheter til egne test-metadata (organisasjonen slettes?). Behovet for å kunne slette hele katalogen gjelder kun demo-miljøet - ikke skarp løsning. Eksempler på scenarier der dette blir et behov:

1. Bruker har høstet in test-data, men ønsker å fjerne alt å begynne på nytt med ny test
2. Bruker har optimalisert innhøstingen og er ferdig med testing

Forretningsverdi (Kritisk/Viktig/Mindre viktig/*Kjekt å ha*):

Lenker (hvis ikke relevant, skriv 'N/A')
Lenke til design:
Lenke til løsningsarkitektur:
Lenke til trusselmodellering:

### Avklaringer: 

| Spørsmål | Avklaring | Avklart med | Avklart dato | 
| -------- | --------- | ----------- | ------------ |
|Gjelder dette bare demo-løsning? | Ja, det er kun i testfasen dette er et behov | nn | dd|
|Skal slettingen godkjennes av andre? | Nei, dette er bare testdata | nn | dd |
|Skal alt innhold i katalogen slettes? | Ja | nn | dd |
|Skal selve katalogen også slettes? | Ja | nn | dd |
|Gjelder dette både høste- og registreringsløsning? | Kun høsteløsning | nn | dd |
|Gjelder dette alle typer kataloger? | Ja | nn | dd |

### Akseptansekriterier:

1. Gitt at brukeren velger å slette katalogen, så skal
  a) katalog-nivået slettes
  b) alt innhold i katalogen (datasett) slettes
  c) organisasjonen beholdes
  d) ny katalog skal kunne opprettes

**Comment by KjerstiSteien** [comment]
Denne saken bør brytes ned i mindre, håndterbare biter.

**Comment by KurtStian** [comment]
@oystein-asnes Kan du gi mer informasjon om denne? Gjelder det f.eks. sletting av høsteendepunkt i høsteadmin.

**Comment by KurtStian** [comment]
Vi trenger mer informasjon, men dette er sannsynligvis en omfattende jobb.

**Comment by oystein-asnes** [comment]
> @oystein-asnes Kan du gi mer informasjon om denne? Gjelder det f.eks. sletting av høsteendepunkt i høsteadmin.

Har beskrevet mer, men er litt forvirret. Sletting av høsteendepunkt er allerede en funksjon i høsteløsningen. Gitt at den fjerner alt innhold i vår database (altså katalogen og dets bestanddeler), så er denne allerede fikset @KurtStian ?

**Comment by KurtStian** [comment]
@oystein-asnes Ja, det stemmer at sletting av høsteendepunkt allerede er en funksjon. Men sletting av høsteendepunktet fjerner ikke det som allerede er innhøstet. Og som du skriver lenger opp vil en sletting av høsteendepunktet i dag medføre at virksomheten mister muligheten til å oppdatere/"fjerne" sine data (dvs. han kan legge inn det samme høsteendepunktet på nytt og få tilgang igjen). 
Ved å f.eks. fjerne relasjonene til datasettene i klassen katalog til en datasettkatalog vil datasettene blir fjernet fra søkeindeksen slik at de ikke lenger vises i portalen. Men de vil fremdeles være tilgjengelig hvis man har URL-ene.

**Comment by KjerstiSteien** [comment]
Burde tittel og brukerhistorie endres for å tydeliggjøre at det er snakk om en forbedring av funksjonen sletting av høsteendepunkt?

**Comment by oystein-asnes** [comment]
> Burde tittel og brukerhistorie endres for å tydeliggjøre at det er snakk om en forbedring av funksjonen sletting av høsteendepunkt?

Done. Har en mistanke om at dette behovet også er gyldig for prod. (ikke bare demo).

**Comment by oystein-asnes** [comment]
> @oystein-asnes Ja, det stemmer at sletting av høsteendepunkt allerede er en funksjon. Men sletting av høsteendepunktet fjerner ikke det som allerede er innhøstet. Og som du skriver lenger opp vil en sletting av høsteendepunktet i dag medføre at virksomheten mister muligheten til å oppdatere/"fjerne" sine data (dvs. han kan legge inn det samme høsteendepunktet på nytt og få tilgang igjen). Ved å f.eks. fjerne relasjonene til datasettene i klassen katalog til en datasettkatalog vil datasettene blir fjernet fra søkeindeksen slik at de ikke lenger vises i portalen. Men de vil fremdeles være tilgjengelig hvis man har URL-ene.

 Hvorfor fjernes ikke katalogen i dag? Når er det behov for å slette endepunktet og ikke katalogen?

Har endret tittel og justert brukerhistorien ettersom dette uansett ikke er en ny funksjon med endring av slettefunksjonen som alt ligger der.

**Comment by AinaEngen** [comment]
Venter med denne til etter sommeren
